parity/rpc_apis.rs

```diff
@@ -152,21 +180,41 @@ fn to_modules(apis: &[Api]) -> BTreeMap<String, String> {
 }
 
 impl ApiSet {
+	/// Retains only APIs in given set.
+	pub fn retain(self, set: Self) -> Self {
+		ApiSet::List(&self.list_apis() & &set.list_apis())
+	}
+
 	pub fn list_apis(&self) -> HashSet<Api> {
-		let mut safe_list = vec![Api::Web3, Api::Net, Api::Eth, Api::Parity, Api::Traces, Api::Rpc]
-			.into_iter().collect();
+		let mut public_list = vec![
+			Api::Web3, Api::Net, Api::Eth, Api::Parity, Api::Rpc,
+		].into_iter().collect();
```

Comment:
in places like these I wish we had an "iter" macro in the standard library: http://play.integer32.com/?gist=d0bf3fcbfcbb35ad8752ed5be30bda64&version=undefined